File: algorithms/0240.search_a_2d_matrix_ii/search_a_2d_matrix_ii.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        const int n = matrix.size(), m = matrix[0].size();
        int x = 0, y = m - 1;
        while (x < n && y >= 0) {
            if (matrix[x][y] < target) {
                x += 1;
            } else if (matrix[x][y] > target) {
                y -= 1;
            } else {
                return true;
            }
        }
        return false;
    }
};

File: algorithms/0240.search_a_2d_matrix_ii/search_a_2d_matrix_ii.rs
impl Solution {
    pub fn search_matrix(matrix: Vec<Vec<i32>>, target: i32) -> bool {
        if matrix.is_empty() {
            return false;
        }
        let (mut x, mut y) = (0, matrix[0].len() as i32 - 1);
        while x < matrix.len() && y >= 0 {
            if matrix[x][y as usize] == target {
                return true;
            } else if matrix[x][y as usize] > target {
                y -= 1;
            } else {
                x += 1;
            }
        }
        false
    }
}
